services/retentionmgr/retention.go

```diff
@@ -312,26 +314,117 @@ func (t *RetentionManager) runRetention(jobsC chan<- *retentionJob) bool {
 			continue
 		}
 
+		log := t.logger.WithField(common.TagDst, dest.id)
```

Comment:
Is it possible to UT the new behavior

Reply:
it can be .. though it will be quite some work; will give it a shot.

Reply:
as discussed offline, i am going to take this up separately.